src/Symfony/Component/DependencyInjection/ReverseContainer.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+
+/**
+ * Turns public and "container.reversible" services back to their ids.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class ReverseContainer
+{
+    private $serviceContainer;
+    private $reversibleLocator;
+    private $getServiceId;
+
+    public function __construct(Container $serviceContainer, ContainerInterface $reversibleLocator, string $tagName = 'container.reversible')
+    {
+        $this->serviceContainer = $serviceContainer;
+        $this->reversibleLocator = $reversibleLocator;
+        $this->getServiceId = \Closure::bind(function ($service) {
+            return array_search($service, $this->services, true) ?: array_search($service, $this->privates, true);
+        }, $serviceContainer, Container::class);
+    }
+
+    /**
+     * Returns the id of the passed object when it exists as a service.
+     *
+     * To be reversible, services need to be either public of be tagged with "container.reversible".
+     *
+     * @param object $service
+     */
+    public function getId($service): ?string
+    {
+        if ($this->serviceContainer === $service) {
+            return 'service_container';
+        }
+
+        if (null === $id = ($this->getServiceId)($service)) {
```

Comment:
The closure stored in `$this->getService` will return `false` when the service is not found, won't it?

Reply:
correct, fixed